### ISSUE TITLE
Document `hidden` question type

### ISSUE THREAD

**Issue by tinok** [XLSForm/xlsform.github.io#116]
The `hidden` question type is not documented on the main page or the ref table. It refers to [this line](https://github.com/XLSForm/pyxform/blob/3d66f39632414584a5feeedfaf606d038d34d3fc/pyxform/question_type_dictionary.py#L759-L763) in pyxform.

The feature includes a question in the form but doesn't display it. Unfortunately it replaces the actual question type (e.g. `select_one animals`) so it's a bit unclear why it was implemented this way or what the ideal use case is. 

Any thoughts @MartijnR @yanokwa ?

**Comment by tinok** [comment]
It's an important use case to be able to include questions in your XLS form that you don't want to include in the current survey (or even in the XML). 

A better route for a new enhancement would be to add a flag to questions that leads them to being ignored by pyxform during the conversion. E.g. a column `exclude`, and marking each line to be excluded as `TRUE`.

**Comment by MartijnR** [comment]
Wrt to the existing 'hidden' type, I believe the only difference is that the hidden type doesn't require a calculation. 

When I look at the [medic mobile comment](https://github.com/medic/medic-conf/issues/56) linked to PR #96, they like using the `hidden` type to store constants in the XForm (using the `default` column) but without having to use (and explain to users): `calculate="."`.

**Comment by yanokwa** [comment]
And it looks like `hidden` is really a new question to prevent the missing calculation warning. Sounds like the things we used to do in the dark days. Would it be reasonable to get pyxforms to make this more of a macro (so if you use it, it's a `calculate` with `.` as a calculation?

**Comment by yanokwa** [comment]
Fair points. I'm OK with documenting `hidden` with an alias of `constant` (or rather `constant` with an alias of `hidden`). What say ye, @lognaturel?

**Comment by tiritea** [comment]
> It seems to just leave out the calculate requirement, so it's useful to store constants and provides a cleaner XForm output.

Precisely. The purpose of hidden - in my view - is precisely as a means to insert a constant (default) into the XForms instance XML, as @MartijnR  noted. Having to do this using a redundant calculate is inefficient - it'll become yet another calculation to worry about re-evaluating whenever things change...

I actually like hidden exactly the way it is! :-) [although the term 'constant' might indeed be a bit more descriptive]

**Comment by MartijnR** [comment]
* [ ] Slightly elaborate on current description of 'hidden' in docs

**Comment by MartijnR** [comment]
So, this thread was about 2 different things, one of which is addressed.

The remaining is a suggestion about adding a new 'exclude' column. Will update the title.

**Comment by MartijnR** [comment]
Might be nice to support exclude on groups and repeats as well.

**Comment by jnm** [comment]
In case it's helpful for reference: there's an undocumented—except for some SurveyCTO [documentation](https://www.surveycto.com/odk-errors/unmatched-begin-or-end-groups/) and sample forms—`disabled` column that [causes pyxform to completely discard](https://github.com/XLSForm/pyxform/blob/50921a8e48b853565abd48b543558b5e24d6c4f4/pyxform/xls2json.py#L602-L612) rows of the input XLSForm spreadsheet. I found this while [troubleshooting](https://community.kobotoolbox.org/t/error-message-seen-while-trying-to-load-custom-report-server-error-500/10115/16) a problem with a KoBo library that didn't support `disabled`.

It looks like the feature was [added in 2011](https://github.com/XLSForm/pyxform/commit/de45095048cdf231d851a2f0052285891703a531#diff-b29aeb282b118614b8333cdfc19f700dR243-R249) but incurred a [warning later the same year](https://github.com/XLSForm/pyxform/commit/76c1847414e10b8337ca9806408a3e0dbd2e1268#diff-b29aeb282b118614b8333cdfc19f700dR361-L231).
